### PR TITLE
[IMP] web: convert utils test suites to hoot

### PR DESCRIPTION
**Pull request by mcm-odoo**
This PR converts a few utils test suites to hoot: numbers, objects, search, strings, timing and urls.

### CODE REVIEW

**Review by juliusc2066 — APPROVED** [review]

**Comment by mcm-odoo** [comment]
@robodoo delegate=juliusc2066

File: addons/web/static/tests/core/utils/timing.test.js [hunk 1]
```diff
@@ -0,0 +1,525 @@
...
+            () => n++,
+            () => nextAnimationFrame()
+        );
+        expect(n).toBe(0);
+
+        fn();
+        fn();
+        expect(n).toBe(0);
+
+        await nextAnimationFrame();
```
Comment (aab-odoo) [inline addons/web/static/tests/core/utils/timing.test.js:59]: would assert that after a micro tick, n is still 0

**Review by aab-odoo — COMMENTED** [review]
Hello @odoo/rd-security 

Could you please override the CI security? The spotted location object is a mock, it's a plain object used in a qunit test. Thanks

**Comment by aab-odoo** [comment]
robodoo r+

**Comment by Polymorphe57** [comment]
@robodoo rebase-ff

**Pull request closed without merging**